client/dive-common/components/Viewer.vue

```diff
@@ -335,54 +336,59 @@ export default defineComponent({
     );
 
     /** Trigger data load */
-    Promise.all([
-      loadMetadata(props.id).then((meta) => {
-        populateTypeStyles(meta.customTypeStyling);
-        if (meta.customTypeStyling) {
-          importTypes(Object.keys(meta.customTypeStyling), false);
-        }
-        if (meta.attributes) {
-          loadAttributes(meta.attributes);
-        }
-        populateConfidenceFilters(meta.confidenceFilters);
-        datasetName.value = meta.name;
-        frameRate.value = meta.fps;
-        imageData.value = cloneDeep(meta.imageData) as FrameImage[];
-        videoUrl.value = meta.videoUrl;
-        datasetType.value = meta.type as DatasetType; // TODO: support for multiCam will remove this
-      }),
-      loadDetections(props.id).then(async (trackData) => {
-        const tracks = Object.values(trackData);
-        progress.total = tracks.length;
-        for (let i = 0; i < tracks.length; i += 1) {
-          if (i % 4000 === 0) {
+    const loadData = async () => (
```

Comment:
I would like to proclaim there was some thought but I think it's an unconscious thing that if a function is going to be tied to a variable I like to tie to an arrow function unless there is a reason it shouldn't be.

Reply:
Just looking for a guideline for style consistency.  Do you think functions declared in setup() should be arrow functions?

Reply:
I would think so unless you're specifically creating a function object which needs to have a `this` context associated with it.  I would assume though that all functions inside of setup you want to keep fairly clean and only interacting with the components variables and refs to dictate state?  I'm trying to think of why you would want a function with it's own internal  variable like a counter that isn't referenced by the component and keep ending up that it should be a separate imported object/class/function instead of inside of the setup of the component.